source/Ship.cpp

```diff
@@ -3767,6 +3793,62 @@ const vector<weak_ptr<Ship>> &Ship::GetEscorts() const
 
 
 
+// Register ship based conditions.
+void Ship::RegisterDerivedConditions()
+{
+	// Read-only conditions.
+	auto &&hullProvider = conditions.GetProviderNamed("hull");
+	hullProvider.SetGetFunction([this](const string &name) { return hull; });
+
+	auto &&shieldsProvider = conditions.GetProviderNamed("shields");
+	shieldsProvider.SetGetFunction([this](const string &name) { return shields; });
+
+	auto &&outfitInstalledProvider = conditions.GetProviderPrefixed("outfit (installed): ");
+	outfitInstalledProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		const Outfit *outfit = GameData::Outfits().Find(name.substr(strlen("outfit (installed): ")));
+		if(!outfit)
+			return 0;
+		return OutfitCount(outfit);
+	});
+
+	auto &&jumpUsingProvider = conditions.GetProviderPrefixed("jump (using): ");
+	jumpUsingProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		const Outfit *outfit = GameData::Outfits().Find(name.substr(strlen("jump (using): ")));
+		if(!outfit)
+			return 0;
+		if(targetSystem)
+		{
+			pair<JumpType, double> jumpUsed = navigation.GetCheapestJumpType(targetSystem);
+			bool jumpDrive = (jumpUsed.first == JumpType::JUMP_DRIVE) && outfit->TrueName() == "Jump Drive";
+			bool hyperDrive = (jumpUsed.first == JumpType::HYPERDRIVE) &&
+								(outfit->TrueName() == "Hyperdrive" || outfit->TrueName() == "Scram Drive");
```

Comment:
```suggestion
				(outfit->TrueName() == "Hyperdrive" || outfit->TrueName() == "Scram Drive");
```
Also does this account for plugin outfits?

Reply:
Good point, the only real check being done here is the type of jump, so maybe I don't need to define the actual drive being used. I will rewrite this to not require the type of drive.